cookbooks/aws-parallelcluster-install/recipes/c_states.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster
+# Recipe:: c_states
+#
+# Copyright:: 2013-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+return unless node['kernel']['machine'] == 'x86_64'
+
+add_to_list 'Add "processor.max_cstate=1 intel_idle.max_cstate=1" to /etc/default/grub GRUB_CMDLINE_LINUX_DEFAULT' do
```

Comment:
What happens if the entries are already there?

Are there other previous entries in the GRUB_CMDLINE_LINUX_DEFAULT variable? Did you consider `replace_or_add`?

Reply:
Yes, I have considered `replace_or_add` but it do not fit our case